scripts/globals/roe.lua

```diff
@@ -0,0 +1,628 @@
+                if params.dmg and params.dmg > 0 then
+                    local progress = player:getEminenceProgress(697)
+                    player:setEminenceProgress(697, progress + params.dmg)
+                    return true
+                end
+                return false
+            end
+    },
+
+    [45  ] = { -- Weapon Skills 1
+        trigger = triggers.wsuse,
+        goal = 100,
+        reward = { sparks = 500, xp = 2500 },
+    },
+
+  --------------------------------------------
+  -- Crafting: General                      --
+  --------------------------------------------
+
+    [57  ] = { -- Total Successful Synthesis Attempts
+        trigger = triggers.synthsuccess,
+        goal = 30,
+        reward = { sparks = 100, xp = 500, unity = 10, repeatable = true },
+    },
+
+  --------------------------------------------
+  -- Combat (Wide Area) -> Spoils I         --
+  --------------------------------------------
+
+    [71  ] = { -- Spoils - Fire Crystals
+        trigger = triggers.lootitem,
+        goal = 10,
+        reqs = { itemid = set{ 4096 } },
+        reward = { sparks = 200, xp = 1000, unity = 20, repeatable = true },
+    },
+
+    [72  ] = { -- Spoils - Ice Crystals
+        trigger = triggers.lootitem,
+        goal = 10,
+        reqs = { itemid = set{ 4097 } },
+        reward = { sparks = 200, xp = 1000, unity = 20, repeatable = true },
+    },
+
+    [73  ] = { -- Spoils - Wind Crystals
+        trigger = triggers.lootitem,
+        goal = 10,
+        reqs = { itemid = set{ 4098 } },
+        reward = { sparks = 200, xp = 1000, unity = 20, repeatable = true },
+    },
+
+    [74  ] = { -- Spoils - Earth Crystals
+        trigger = triggers.lootitem,
+        goal = 10,
+        reqs = { itemid = set{ 4099 } },
+        reward = { sparks = 200, xp = 1000, unity = 20, repeatable = true },
+    },
+
+    [75  ] = { -- Spoils - Lightning Crystals
+        trigger = triggers.lootitem,
+        goal = 10,
+        reqs = { itemid = set{ 4100 } },
+        reward = { sparks = 200, xp = 1000, unity = 20, repeatable = true },
+    },
+
+    [76  ] = { -- Spoils - Water Crystals
+        trigger = triggers.lootitem,
+        goal = 10,
+        reqs = { itemid = set{ 4101 } },
+        reward = { sparks = 200, xp = 1000, unity = 20, repeatable = true },
+    },
+
+    [77  ] = { -- Spoils - Light Crystals
+        trigger = triggers.lootitem,
+        goal = 10,
+        reqs = { itemid = set{ 4102 } },
+        reward = { sparks = 200, xp = 1000, unity = 20, repeatable = true },
+    },
+
+    [78  ] = { -- Spoils - Dark Crystals
+        trigger = triggers.lootitem,
+        goal = 10,
+        reqs = { itemid = set{ 4103 } },
+        reward = { sparks = 200, xp = 1000, unity = 20, repeatable = true },
+    },
+    
+  ----------------------------------------
+  -- Combat (Region) - Original Areas 2 --
+  ----------------------------------------
+
+    [239 ] = { -- Conflict: Jugner Forest
```

Comment:
I don't know what options you've explored for this, but how possible is it to move these sorts of lookups into the `Zone.lua` for these areas to reduce bloat?

Reply:
As the records list gets bigger, I may certainly part out the main record list into its own Lua file, (ala how Magian does) but for now the list was small enough. :+1: